rapid/src/main/java/com/vrg/rapid/MembershipService.java

```diff
@@ -228,55 +197,54 @@
+                            .setConfigurationId(configuration.getConfigurationId())
+                            .setNodeId(joinMessage.getNodeId())
+                            .addRingNumber(i)
+                            .setMetadata(joinMessage.getMetadata())
+                            .build();
+                    enqueueAlertMessage(msg);
+                }
             } else {
-                // This handles the corner case where the configuration changed between phase 1 and phase 2
-                // of the joining node's bootstrap. It should attempt to rejoin the network.
-                final MembershipView.Configuration configuration = membershipView.getConfiguration();
-                LOG.info("Wrong configuration for {sender:{}, config:{}, myConfig:{}, size:{}}",
-                        Utils.loggable(joinMessage.getSender()), joinMessage.getConfigurationId(),
-                        currentConfiguration, membershipView.getMembershipSize());
-                JoinResponse.Builder responseBuilder = JoinResponse.newBuilder()
+                // no.
+                final JoinResponse response = JoinResponse.newBuilder()
                         .setSender(myAddr)
-                        .setConfigurationId(configuration.getConfigurationId());
-                if (membershipView.isHostPresent(joinMessage.getSender())
-                        && membershipView.isIdentifierPresent(joinMessage.getNodeId())) {
-                    LOG.info("Joining host already present : {sender:{}, config:{}, myConfig:{}, size:{}}",
-                            Utils.loggable(joinMessage.getSender()), joinMessage.getConfigurationId(),
-                            currentConfiguration, membershipView.getMembershipSize());
-                    // Race condition where a observer already crossed H messages for the joiner and changed
-                    // the configuration, but the JoinPhase2 messages show up at the observer
-                    // after it has already added the joiner. In this case, we simply
-                    // tell the sender that they're safe to join.
-                    responseBuilder = responseBuilder.setStatusCode(JoinStatusCode.SAFE_TO_JOIN)
-                            .addAllEndpoints(configuration.endpoints)
-                            .addAllIdentifiers(configuration.nodeIds)
-                            .putAllClusterMetadata(metadataManager.getAllMetadata());
-                } else {
-                    responseBuilder = responseBuilder.setStatusCode(JoinStatusCode.CONFIG_CHANGED);
-                    LOG.info("Returning CONFIG_CHANGED for {sender:{}, config:{}, size:{}}",
-                            Utils.loggable(joinMessage.getSender()), configuration.getConfigurationId(),
-                            configuration.endpoints.size());
-                }
-                future.set(Utils.toRapidResponse(responseBuilder.build())); // new configuration
+                        .setStatusCode(statusCode)
```

Comment:
The comment says "no", so I assume you want to reject the join attempt here. But the statusCode has two other values it could take. Might be a good time to switch this if-else-if-else into a switch statement to be explicit about what we are returning here.